src/Cli/dotnet/commands/dotnet-workload/install/NetSdkManagedInstaller.cs

```diff
@@ -37,7 +38,7 @@ internal class NetSdkManagedInstaller : IWorkloadPackInstaller
             INuGetPackageDownloader nugetPackageDownloader = null,
             string dotnetDir =  null)
         {
-            _dotnetDir = dotnetDir ?? EnvironmentProvider.GetDotnetExeDirectory();
+            _dotnetDir = dotnetDir ?? Path.GetDirectoryName(Process.GetCurrentProcess().MainModule.FileName);
```

Comment:
nit: currentProcess is disposable, and there is a much fast API available in 6.0 onwards.
```suggestion
#if NET6_0
            // fast path
            _dotnetDir = dotnetDir ?? Path.GetDirectoryName(Environment.ProcessPath);
 #else
             // slow path
             using Process process = Process.GetCurrentProcess();
             _dotnetDir = dotnetDir ?? Path.GetDirectoryName(process.MainModule.FileName);
 #endif
```

Reply:
Thanks for the suggestion, personally I prefer the simplicity of the single line over the small perf change, especially since this is not a perf critical command.

Reply:
I agree .NET 6 API is perf suggestion.
However, not disposing the disposable object is the main problem I have highlighted.